docs/coordinates/index.rst

```diff
@@ -425,6 +425,56 @@ and transforming velocities. These are available both via the lower-level
 For more details on velocity support (and limitations), see the
 :doc:`velocities` page.
 
+.. _astropy-coordinates-masks:
+
+Masks
+-----
+
+Sometimes you may have incomplete information about objects, e.g., some have
+distances while others have not. `~astropy.coordinates` supports using masks
+for such purposes, using the |Masked| class::
+
+    >>> from astropy.utils.masked import Masked
+    >>> distance = Masked([0.1, np.nan]*u.kpc, mask=[False, True])
+    >>> sc = SkyCoord([1., 2.]*u.hourangle, [3., 4.]*u.deg, distance=distance)
+    >>> sc
+    <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, kpc)
+        [(15., 3., 0.1), (30., 4., ———)]>
+
+The masks propagates as you would expect::
+
+    >>> sc.separation(sc[0])  # doctest: +FLOAT_CMP
+    <MaskedAngle [ 0.        , 15.00502838] deg>
+    >>> sc.separation_3d(sc[0])  # doctest: +FLOAT_CMP
+    <MaskedDistance [ 0., ———] kpc>
+    >>> sc.fk5  # doctest: +FLOAT_CMP
+    <SkyCoord (FK5: equinox=J2000.000): (ra, dec, distance) in (deg, deg, kpc)
+        [(15.00000668, 3.00000101, 0.1), (        ———,        ———, ———)]>
+
+In the last example, you will notice that the angles of the second item have
+become masked too. This is because the distance is used in the conversion
```

Comment:
Not for this PR, but for frame transforms that are only rotations we could avoid going to the cartesian representation, instead doing change of basis stuff on the rotation matrix.

Reply:
The problematic code is https://github.com/astropy/astropy/blob/1b82175461b87745e5328fc1622ee8e30b38fa1b/astropy/coordinates/transformations/affine.py#L118-L131

I tried a simple removal of the conversion to cartesian, since the matrix multiplication can be handled by the different representation classes internally. Unfortunately, adding the offset right now does require moving to cartesian. Obviously, this is solvable too, but best in a different PR!

Reply:
Yes, sounds like a good idea to change the transformation example -- and maybe mention in the general warning about this being new that sometimes masks may propagate further than expected.

Reply:
We should open an issue so that any users who might want to report this wouldn't have to write one up themselves, but doing that can wait until the feature freeze has happened.